Firestore/Source/API/FIRDocumentSnapshot.mm

```diff
@@ -191,27 +171,24 @@ - (nullable id)valueForField:(id)field
     FSTThrowInvalidArgument(@"Subscript key must be an NSString or FIRFieldPath.");
   }
 
-  FSTFieldValue *fieldValue = [[self.internalDocument data] valueForPath:fieldPath.internalValue];
-  FSTFieldValueOptions *options = [self optionsForServerTimestampBehavior:serverTimestampBehavior];
+  FSTFieldValue *fieldValue = _shim.GetValue(fieldPath.internalValue);
   return fieldValue == nil ? nil : [self convertedValue:fieldValue options:options];
 }
 
-- (FSTFieldValueOptions *)optionsForServerTimestampBehavior:
+- (nullable id)objectForKeyedSubscript:(id)key {
+  return [self valueForField:key];
+}
+
+- (FSTFieldValueOptions *)createOptionsForServerTimestampBehavior:
```

Comment:
`optionsForServerTimestampBehavior` was the better name for this according to the [style guide](http://google.github.io/styleguide/objcguide.html#objective-c-method-names) (see the `sandwich` example). Please revert.